app/scripts/lib/util.ts

```diff
@@ -433,3 +435,38 @@ export const getMethodDataName = async (
 export function getBooleanFlag(value: string | boolean | undefined): boolean {
   return value === true || value === 'true';
 }
+
+type AssetsRatesState = {
+  metamask: MultichainAssetsRatesControllerState;
+};
+
+export function getConversionRatesForNativeAsset({
+  conversionRates,
+  chainId,
+}: {
+  conversionRates: AssetsRatesState['metamask']['conversionRates'];
+  chainId: string;
+}): { rate: number } | null {
+  // Return early if conversionRates is falsy
+  if (!conversionRates) {
+    return null;
+  }
+
+  let conversionRateResult = null;
+
+  Object.entries(conversionRates).forEach(
+    ([caip19Identifier, conversionRate]) => {
+      const parsedCaip19Identifier = parseCaipAssetType(
+        caip19Identifier as CaipAssetType,
```

Comment:
Why do we need to cast here ? The type should be good on the controller side

Reply:
I think the cast was needed here because using `Object.entries()` in TypeScript does not preserve the original type of the object. The key was generalized as string, without the cast you would get : 
`Argument of type 'string' is not assignable to parameter of type '`${string}:${string}/${string}:${string}`'`

I think its fine to keep the cast here